server/aesdsocket.c

```diff
@@ -0,0 +1,183 @@
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <errno.h>
+#include <syslog.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+
+#define PORT "9000"
+#define BACKLOG 10
+#define BUF_LEN 1000000
+
+char *pathname = "/var/tmp/aesdsocketdata";
+int sockfd;
+
+// get sockaddr, IPv4 or IPv6:
+void *get_in_addr(struct sockaddr *sa) {
+    if (sa->sa_family == AF_INET) {
+        return &(((struct sockaddr_in*)sa)->sin_addr);
+    }
+    return &(((struct sockaddr_in6*)sa)->sin6_addr);
+}
+
+static void signal_handler(int signo) {
+    syslog(LOG_INFO, "Caught Signal, exiting\n");
+    close(sockfd);
+    remove(pathname);
+    closelog();
+    exit(EXIT_SUCCESS);
```

Comment:
@ashwinhr186 are all of these safe to do in a signal handler?  Consider a design where you can handle cleanup outside your signal handler.